icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -642,9 +613,10 @@ SimpleDateFormat& SimpleDateFormat::operator=(const SimpleDateFormat& other)
     }
 
     UErrorCode localStatus = U_ZERO_ERROR;
-    freeFastNumberFormatters();
-    initFastNumberFormatters(localStatus);
-
+    // SimpleNumberFormatter does not have a copy constructor. Furthermore,
+    // it references data from an internal field, fNumberFormatter,
+    // so we must rematerialize that reference after copying over the number formatter
```

Comment:
FYI, maybe for future PRs, and ICU style is not strict:
I like the google3 style where if a comment contains a complete sentence it should have a terminating period.